src/App.tsx

```diff
@@ -16,60 +15,150 @@ const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
+  NONE = 'none',
+  ALPABET = 'alphabet',
+  LENGTH = 'length',
 }
 
-// Use this function in the render method
 function getReorderedGoods(
   goods: string[],
   sortType: SortType,
   isReversed: boolean,
+  minLength: number,
 ) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
+  const visibleGoods = [...goods].filter(element => (
+    element.length >= minLength
+  ));
 
-  // Sort and reverse goods if needed
-  // ...
+  if (sortType === SortType.ALPABET) {
+    visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+  } else if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
 
-  return visibleGoods;
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
```

Comment:
perhaps something like switch case? leaving `else if` as the last condition is not a common practice